internal/pkg/upgrade/upgrade.go

```diff
@@ -27,6 +32,21 @@ func NewUpgrade(url string) (err error) {
 		return
 	}
 
+	data, err := userdata.Open(constants.UserDataPath)
+	if err != nil {
+		return err
+	}
+	if data, err = data.Upgrade(); err != nil {
+		return err
+	}
+	dataBytes, err := yaml.Marshal(data)
```

Comment:
Feel like this would be more appropriate as a method off userdata ( `func (u *UserData) Write()` )

Reply:
I agree. I feel that is another PR, since we use the method I am in a number of places. I will follow up with a PR.